index.bs

```diff
@@ -10791,6 +10576,174 @@ property-installation style as namespaces.)
+        1.  Let |id| be the [=identifier=] of |attr|.
+        1.  Let |value| be the underlying value of |attr|.
+        1.  Let |value| be the result of performing the actions listed in the
+            description of |attr| that occur on getting (or those listed in the description
+            of the inherited attribute, if this attribute is declared to
+            [=inherit its getter=]), on |O|.
+        1.  If |value| is a [=JSON type=], then [=map/set=] |map|[|id|] to |value|.
+    1.  If |stack| [=stack/is not empty=],
+        then invoke [=collect attribute values=] on |O|,
+        passing it |stack| and |map| as arguments.
+</div>
+
+<div algorithm>
+
+    To <dfn>create an inheritance stack</dfn> for [=interface=] |I|,
+    run the the following steps:
+
+    1.  Let |stack| be a new [=stack=].
+    1.  [=stack/Push=] |I| onto |stack|.
+    1.  [=iteration/While=] |I| [=interface/inherits=] from an [=interface=],
+        1.  Let |I| be that [=interface=].
+        1.  [=stack/Push=] |I| onto |stack|.
+    1.  Return |stack|.
+</div>
+
+<div class=example>
+
+    The following [=IDL fragment=] defines a number of [=interfaces=]
+    which are [=inherited interface|inherited=] or [=consequential interfaces=] of `A`,
+    as show in the below inheritance tree.
+
+    <pre>
+                  F*
+                  |
+             C    E - I*
+             |    |
+             B* - D*
+             |
+        G* - A - H*
+    </pre>
+
+    Interfaces markes with an asterixk ("*")
+    declare a <code>toJSON</code> [=operation=]
+    with a [{{Default}}] [=extended attribute=].
+
+    <pre highlight=webidl>
+        interface A : B {
+          attribute DOMString r;
+        };
+
+        interface B : C {
+          [Default] object toJSON();
+          attribute DOMString s;
+        };
+
+        interface C {
+          attribute DOMString t;
+        };
+
+        interface D : E {
+          [Default] object toJSON();
+          attribute DOMString u;
+        };
+
+        interface E : F {
+          attribute DOMString v;
+        };
+
+        interface F {
+          [Default] object toJSON();
+          attribute DOMString w;
+        };
+
+        interface G {
+          [Default] object toJSON();
+          attribute DOMString x;
+        };
+
+        interface H {
+          [Default] object toJSON();
+          attribute DOMString y;
+        };
+
+        interface I {
+          [Default] object toJSON();
+          attribute DOMString z;
+        };
+
+        A implements G;
+        A implements H;
+        B implements D;
+        E implements I;
+    </pre>
+
+    Calling the <code>toJSON</code> method of an object
+    implementing interface `A` defined above
+    would return the following JSON object:
+
+    <pre highlight=json>
+    {
+        "w": "...",
+        "z": "...",
+        "u": "...",
+        "s": "...",
+        "x": "...",
+        "y": "..."
+    }
```

Comment:
OK, so it would be easier to check this if the lowercase property names matched the uppercase interface names.

So, this order is a bit surprising to me. I would the following interface order:

> C, B, F, E, I, D, A, G, H

My reasoning is:

- First go to the most distance ancestor on the JavaScript prototype chain (C).
- Then go to the next-most-distance, B
- Then go B's mixins
  - But D has inheritance, so go up its "chain", ending up with F; then E plus its I mixin; then finally D
- Now we can do A, and then its two mixins, G and H.

---

This all might be moot once we simplify interfaces... but I think it'd be good to get right, if possible. We could add a warning that we anticipate such complex inherited/consequential interface chains not being possible in the future, which would make all this less ridiculous. Probably linking to the mixin issue I'm about to open.